### PR TITLE
fix(InputItem): money keyboard throw error when page restore

### PR DESCRIPTION
**Pull request by paranoidjk**
Fix https://github.com/ant-design/ant-design-mobile/issues/2207

Fix https://github.com/ant-design/ant-design-mobile/issues/2175

First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [ ] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [ ] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [ ] Rebase before creating a PR to keep commit history clear.
* [ ] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ ] Update API docs for the component.
  * [ ] Update/Add demo to demonstrate new feature.
  * [ ] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

---
This change is [Reviewable](https://reviewable.io/reviews/ant-design/ant-design-mobile/2220)

### CODE REVIEW

**Comment by paranoidjk** [comment]
@silentcloud 也看下，你加了个 isPortal 的全局锁，导致组件 unmount 之后再次 mount 的时候，由于锁的原因不会再渲染 getComponent

File: components/input-item/CustomInput.tsx [hunk 1]
```diff
@@ -78,17 +74,14 @@ class NumberInput extends React.Component<any, any> {
 
   getComponent() {
     const { keyboardPrefixCls, confirmLabel } = this.props;
-    if (!hasPortal || !IS_REACT_16) {
```
Comment (paranoidjk) [inline components/input-item/CustomInput.tsx]: @silentcloud 这里拦截有什么意图？

Comment (silentcloud) [reply components/input-item/CustomInput.tsx]: 我记得应该是多个 money input 的情况

Comment (paranoidjk) [reply components/input-item/CustomInput.tsx]: 我写个多个 input 的 demo 试试

Comment (silentcloud) [reply components/input-item/CustomInput.tsx]: 当时是一个用户反馈的，还要注意 路由跳转新页面在回来的情况

Comment (paranoidjk) [reply components/input-item/CustomInput.tsx]: > 由跳转新页面在回来的情况

这个就是你加了锁导致的bug，可以看上面 issue 里面的例子

Comment (paranoidjk) [reply components/input-item/CustomInput.tsx]: 测了单页面多个实例，以及 SPA 路由来回切换的情况，ok

**Pull request merged**